### ISSUE TITLE
Detect if a Short URL is being passed as a Long URL

### ISSUE THREAD

**Issue by ajtatum** [shlinkio/shlink#942]
I debated between this being a bug or a feature, so I went with feature. One thing I noticed with the sharing plugin that I use (custom), is that when someone clicks to share via Twitter, Facebook, etc I make a call to the API to request a short URL and then set the URL to be shared. If a user clicks the button twice, for whatever reason, it shortens the already short URL. I'm working to try and fix the UI/UX on my end; however, I'm coming from Bitly and with them if you sent a POST request to create a short URL from a short URL, it would return the already existing short URL rather than creating a new short url. 

I hope this makes sense LOL, if not let me know.

Thanks,
AJ

**Comment by acelaya** [comment]
So, let me see if I'm getting what you mean.

You create a short URL by passing, for example `{ "longUrl": "https://www.google.com" }` and the resulting short URL is `https://foo.com/abc123`

Then you try to create a new short URL by passing `{ "longUrl": "https://foo.com/abc123" }`. This currently results in a second short URL being created, that redirects to the first one, resulting in this redirection flow:

`https://foo.com/def456` -> `https://foo.com/abc123` -> `https://www.google.com`.

What you mean is that you would expect the second request to result in `https://foo.com/abc123` too?

**Comment by ajtatum** [comment]
That's correct. Again, I'm working on a work around for my site, but figured I'd submit this as it was something that Bitly handled before.

**Comment by acelaya** [comment]
I think it's a bit of an edge case, and it might be hard to handle properly, because short URLs can have some metadata (max visits, tags, etc).

For example, should it behave the same if the second time you provide some tags while you didn't the first time? Should it add the tags to the original URL instead?

It's trickier than it looks.

**Comment by ajtatum** [comment]
You're right; it is an edge case. I've updated my code (which is based on https://github.com/ellisonleao/sharer.js) to handle it better. Sharer.js uses the data-url parameter to know what URL to share, previously when someone clicked "Share" I would make a call to Bitly, get the Short Url and then update that data-url attribute. Thus, if someone then clicked "Share" again, it would pull the short url from the data-url attribute. I've updated the javascript so that I create a new attribute, data-shorturl, which is what is now "pulled" from Sharer.js to share (thus leaving the long, data-url attribute the same). With Shlink's implementation, I pass in "findIfExists: true" which then generates the same short url. Which is absolutely perfect.
